Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -160,6 +153,266 @@ theorem support_eq_empty {p} : p.support = ∅ ↔ (p : SkewMonoidAlgebra k G) =
+    {a : S} [Nonempty G] : IsSMulRegular k a ↔ IsSMulRegular (SkewMonoidAlgebra k G) a := by
+  inhabit G
+  refine ⟨IsSMulRegular.skewMonoidAlgebra, fun ha b₁ b₂ inj ↦ ?_⟩
+  rw [← (single_injective _).eq_iff, ← smul_single, ← smul_single] at inj
+  exact single_injective (default) (ha inj)
+
+end Single
+
+section One
+
+variable [One G] [One k]
+
+/-- The unit of the multiplication is `single 1 1`, i.e. the function that is `1` at `1` and
+  zero elsewhere. -/
+instance instOne : One (SkewMonoidAlgebra k G) := ⟨single 1 1⟩
+
+theorem ofFinsupp_one : (⟨Finsupp.single 1 1⟩ : SkewMonoidAlgebra k G) = 1 := rfl
+
+@[simp]
+theorem toFinsupp_one : (1 : SkewMonoidAlgebra k G).toFinsupp = Finsupp.single 1 1 := rfl
+
+@[simp]
+theorem toFinsupp_eq_single_one_one_iff {a : SkewMonoidAlgebra k G} :
+    a.toFinsupp = Finsupp.single 1 1 ↔ a = 1 := by
+  rw [← toFinsupp_one, toFinsupp_inj]
+
+@[simp]
+theorem ofFinsupp_eq_one {a} :
+    (⟨a⟩ : SkewMonoidAlgebra k G) = 1 ↔ a = Finsupp.single 1 1 := by
+  rw [← ofFinsupp_one, ofFinsupp_inj]
+
+theorem single_one_one  : single 1 (1 : k) = 1 := rfl
+
+theorem one_def : (1 : SkewMonoidAlgebra k G) = single 1 1 := rfl
+
+@[simp]
+theorem coeff_one_one : coeff (1 : SkewMonoidAlgebra k G) 1 = 1 := by
+  simp only [coeff, toFinsupp_single, Finsupp.single_eq_same]
+
+theorem coeff_one {a : G} [Decidable (a = 1)] :
+    (1 : SkewMonoidAlgebra k G).coeff a = if a = 1 then 1 else 0 := by
+  classical
+  simp_rw [eq_comm (a := (a : G)) (b := (1 : G))]
+  simpa using coeff_single
+
+end One
+
+section sum
+
+instance [DecidableEq G] [DecidableEq k] : DecidableEq (SkewMonoidAlgebra k G) :=
+  Equiv.decidableEq toFinsuppAddEquiv.toEquiv
+
+/-- `sum f g` is the sum of `g a (f.coeff a)` over the support of `f`. -/
+def sum {N : Type*} [AddCommMonoid N] (f : SkewMonoidAlgebra k G) (g : G → k → N) : N :=
+  f.toFinsupp.sum g
+
+theorem sum_def {N : Type*} [AddCommMonoid N] (f : SkewMonoidAlgebra k G) (g : G → k → N) :
+    sum f g = ∑ a ∈ f.support, g a (f.coeff a) := rfl
+
+@[simp]
+theorem sum_single_index {N} [AddCommMonoid N] {a : G} {b : k} {h : G → k → N}
+    (h_zero : h a 0 = 0) : (SkewMonoidAlgebra.single a b).sum h = h a b :=
+  Finsupp.sum_single_index h_zero
+
+theorem map_sum {N P : Type*} [AddCommMonoid N] [AddCommMonoid P] {H : Type*} [FunLike H N P]
+    [AddMonoidHomClass H N P] (h : H) (f : SkewMonoidAlgebra k G) (g : G → k → N) :
+    h (sum f g) = sum f fun (a : G) (b : k) => h (g a b) :=
+  _root_.map_sum h _ _
+
+theorem toFinsupp_sum {N : Type*} [AddCommMonoid N] (f : SkewMonoidAlgebra k G) (g : G → k → N) :
+    sum f g = Finsupp.sum f.toFinsupp g := rfl
+
+/-- Variant where the image of `g` is a `SkewMonoidAlgebra`. -/
+theorem toFinsupp_sum' {k' G' : Type*} [AddCommMonoid k'] (f : SkewMonoidAlgebra k G)
+    (g : G → k → SkewMonoidAlgebra k' G') :
+    (sum f g).toFinsupp = Finsupp.sum f.toFinsupp (fun (a : G) (b : k) ↦ (g a b).toFinsupp) :=
```

Comment:
I think you can feel free to omit the type annotations:
```suggestion
    (sum f g).toFinsupp = Finsupp.sum f.toFinsupp (fun a b ↦ (g a b).toFinsupp) :=
```
I might actually write `(toFinsupp <| g · ·)`.